### PR TITLE
compability parts with MySQL dialect

### PR DESCRIPTION
**Pull request by evlampiy-lavrentiev**
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- added functions
`FuncitonInstr`
`FunctionASCII`
`FunctionBitLength`
`FunctionConcatWs`
`FunctionConv`
`FunctionElt`
`FunctionFormatBytes`
`FunctionOrd`
`FunctionSpace`
- added tests for functions
`FunctionASCII` (edge case handling)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Functions/FunctionFormatBytes.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#include <DataTypes/DataTypeString.h>
```
Comment (alexey-milovidov) [inline src/Functions/FunctionFormatBytes.cpp:1]: We already have a similar function.

File: src/Functions/FunctionElt.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        auto string_type = std::make_shared<DataTypeInt32>();
+        auto casted_column = castColumn(std::move(arguments[0]), string_type);
+        const ColumnInt32 * column_concrete = checkAndGetColumn<ColumnInt32>(casted_column.get());
+        const typename ColumnVector<Int32>::Container & idx_str = column_concrete->getData();
+       
+        auto result_column = ColumnString::create();
+        
+        std::vector<std::string> ans(input_rows_count);
```
Comment (alexey-milovidov) [inline src/Functions/FunctionElt.cpp:51]: This is inefficient.

File: src/Functions/FunctionConv.cpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            String ans_s;
+            base = num_to[i];
+            if (base < 0) {
+                base = -base;
+                neg++;
+            }
+            while (ans > 0) {
+                int64_t add = ans % base;
+                if (add < 10) {
+                    ans_s += toString(add);
```
Comment (alexey-milovidov) [inline src/Functions/FunctionConv.cpp:111]: Inefficient.

File: src/Functions/FunctionConv.cpp [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                ans += add;
+            }
+            String ans_s;
+            base = num_to[i];
+            if (base < 0) {
+                base = -base;
+                neg++;
+            }
+            while (ans > 0) {
+                int64_t add = ans % base;
```
Comment (alexey-milovidov) [inline src/Functions/FunctionConv.cpp:109]: Inefficient.

File: src/Functions/FuncitonInstr.cpp [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+#include <Functions/FunctionHelpers.h>
+#include <Core/Field.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnString.h>
+#include <Interpreters/castColumn.h>
+#include <../src/IO/ReadHelpers.h>
+#include <Functions/FunctionsMultiStringFuzzySearch.h>
+
+
+static UInt32 z_function(String s, String s1) {
```
Comment (alexey-milovidov) [inline src/Functions/FuncitonInstr.cpp:14]: What's this?
